File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/11068_회문인 수/11068.py
def is_palindrome(num, B):
	lst = []
	while num > 0:
		lst.append(num % B)
		num //= B
	
	length = len(lst)

	for i in range(length >> 1):
		if lst[i] != lst[length - i - 1]:
			return 0
	return 1

for _ in range(int(input())):
	num = int(input())
	for B in range(2, 65):
		flag = is_palindrome(num, B)
		if flag == 1:
			break
	print(flag)

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/11068_회문인 수/11068.cpp
#include <stdio.h>
#include <vector>

using namespace std;

int T;
vector<int> vec;

int is_palindrome(int num, int B)
{
	while (num > 0)
	{
		vec.push_back(num % B);
		num /= B;
	}
	
	int size = vec.size();

	for (int i = 0; i < size / 2; ++i)
		if (vec[i] != vec[size - i - 1])
			return 0;

	return 1;
}

int main()
{
	int flag, num;

	scanf(" %d", &T);
	while (T--)
	{
		scanf(" %d", &num);
		for (int B = 2; B <= 64; ++B)
		{
			flag = is_palindrome(num, B);
			if (flag)
				break;
			vec.clear();
		}
		printf("%d\n", flag);
		vec.clear();
	}
	return 0;
}